app/src/routes/api/v1/dataset.router.js

```diff
@@ -260,6 +261,20 @@ class DatasetRouter {
         }
 
         try {
+            if (query.sort && (query.sort.includes('user.role') || query.sort.includes('user.name'))) {
+                logger.debug('Detected sorting by user role or name');
+                if (!user || !['ADMIN', 'SUPERADMIN'].includes(user && user.role)) {
+                    ctx.throw(403, 'Sorting by user name or role not authorized.');
+                    return;
+                }
+                const ids = await DatasetService.getAllDatasetUserIds();
+                const users = await RelationshipsService.getUsersInfoByIds(ids);
+                await Promise.all(users.map(u => DatasetModel.updateMany(
```

Comment:
every time you name a variable with a single letter (even in this type of scenario), god kills a kitten